### PR TITLE
Add prompt to auto import keys and to ignore untrusted keys

### PR DESCRIPTION
**Pull request by ynerant**

### CODE REVIEW

File: default_organisation.bash [hunk 1]
```diff
@@ -86,16 +86,34 @@ set_gpg_recipients() {
...
 				[[ $trust_level =~ [mfu] ]] && [[ $capabilities =~ "e" ]] && can_encrypt=1
 			done < <($GPG --list-key --with-colons "$gpg_id" | grep -E "^sub|^pub")
-			[[ $can_encrypt -eq 0 ]] && gpg --list-key "$gpg_id"
-			[[ $can_encrypt -eq 0 ]] && die "GPG can not encrypt for \"$gpg_id\". Did you import and trust it?"
-
-			GPG_RECIPIENT_ARGS+=( "-r" "$gpg_id" )
-			GPG_RECIPIENTS+=( "$gpg_id" )
+			if [[ $can_encrypt -eq 0 ]] ; then
+				echo ""
+				gpg --list-key "$gpg_id"
```
Comment (erdnaxe) [inline default_organisation.bash]: You should use $GPG rather than gpg. On some system gpg does not map to gpg2.

File: default_organisation.bash [hunk 2]
```diff
@@ -86,16 +86,34 @@ set_gpg_recipients() {
...
+				echo "GPG can not encrypt for \"$gpg_id\". Did you import and trust it?"
+				read -p "Do you want to ignore this fingerprint? [y/N] " input < /dev/tty
```
Comment (erdnaxe) [inline default_organisation.bash]: There is a `yesno` function in pass doing exactly that, but I did not manage to get it working in that context. Did you try?

File: default_organisation.bash [hunk 3]
```diff
@@ -86,16 +86,34 @@ set_gpg_recipients() {
...
+				case $input in
+					[y/Y])	gpg --recv-keys "$gpg_id" ;;
```
Comment (erdnaxe) [inline default_organisation.bash]: Why answering Yes should fetch the key? When answering "yes" you should just skip the key, no?

File: default_organisation.bash [hunk 4]
```diff
@@ -86,16 +86,34 @@ set_gpg_recipients() {
...
 			# Check that gpg know recipient and policy allow to encrypt for them
 			local can_encrypt=0
+			$GPG --list-key "$gpg_id" > /dev/null
+			if ! [[ $? -eq 0 ]]; then
+				echo ""
+				echo "Fingerprint \"$gpg_id\" was not found. Maybe it is not imported."
+				read -r -p "Do you want to import it? [y/N] " input < /dev/tty
+				case $input in
+					[y/Y]) gpg --recv-keys "$gpg_id" || exit 1 ;;
+				esac
```
Comment (erdnaxe) [inline default_organisation.bash]: To keep consistency with `pass`, I would prefer:

```bash
read -r -p "Do you want to import it? [y/N] " response
[[ $response == [yY] ]] && $GPG --recv-keys "$gpg_id" || exit 1
````

Comment (erdnaxe) [reply default_organisation.bash]: Do you need `< /dev/tty` ? This can maybe be an issue on certain platform?

**Review by erdnaxe — CHANGES_REQUESTED** [review]
Nice pull request, LGTM but small details can be changed before merge

File: default_organisation.bash (hunk 4 above)
Comment (ynerant) [reply default_organisation.bash]: Without the `< /dev/tty`, `read` is reading the response from the output of GPG, then the prompt does not appear.

File: default_organisation.bash (hunk 3 above)
Comment (ynerant) [reply default_organisation.bash]: Yes, it is a mistake

File: default_organisation.bash (hunk 2 above)
Comment (ynerant) [reply default_organisation.bash]: I didn't find it, and Internet does not help me...

File: test.bash [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#/bin/bash
+yesno "Coucou"
```
Comment (erdnaxe) [inline test.bash]: Don't commit that please

Comment (erdnaxe) [reply test.bash]: yesno is defined here https://git.zx2c4.com/password-store/tree/src/password-store.sh

**Review by erdnaxe — CHANGES_REQUESTED** [review]
Please remove test.bash (you may squash this with last commit)

File: test.bash (hunk 5 above)
Comment (ynerant) [reply test.bash]: I am an idiot

Comment (erdnaxe) [reply test.bash]: Don't say that, I just did not make explicit enough the fact that this extension is sourced form password-store.sh. I may improve (and write) developer documentation.

**Pull request merged**